File: app/src/main/java/com/iceteaviet/fastfoodfinder/ui/base/BaseActivity.java
package com.iceteaviet.fastfoodfinder.ui.base;

import android.os.Bundle;

import androidx.annotation.LayoutRes;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * Created by tom on 7/10/18.
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
    }

    /**
     * @return layout resource id
     */
    public abstract @LayoutRes
    int getLayoutId();
}

File: app/src/main/java/com/iceteaviet/fastfoodfinder/ui/base/BaseActivity.kt
package com.iceteaviet.fastfoodfinder.ui.base

import javax.inject.Inject
import dagger.hilt.android.AndroidEntryPoint
import android.os.Bundle
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity

/**
 * Created by tom on 7/10/18.
 */
abstract @AndroidEntryPoint
class BaseActivity : AppCompatActivity() {

    /**
     * @return layout resource id
     */
    @get:LayoutRes
    abstract val layoutId: Int

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layoutId)
    }
}
